### PR TITLE
hide plugins screen, and set plugins status to skipped if plugin conf…

### PR DESCRIPTION
**Pull request by neallred**
hide plugins screen, and set plugins status to skipped if plugin config setting is detected as false. If they happen to be on that screen, maybe because they did something odd like disable plugins while on that screen, move them from that screen to the one they used to be on.

#### Summary
If plugins are disabled, then the admin should not be presented with a screen that plugins can be installed in the onboarding.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-42465

#### Related Pull Requests
- Has server changes https://github.com/mattermost/mattermost-server/compare/MM-42465?expand=1

#### Screenshots

User is moved to their most recent page if plugins is disabled. Also, the page is removed from viewable pages:
https://user-images.githubusercontent.com/13738432/157772861-3f226fc6-f754-4d92-89ea-0439c06a0845.mp4

With the related server changes, onboarding/complete endpoint request is successful:

https://user-images.githubusercontent.com/13738432/157772978-03fc90d9-d2f0-45ac-ad48-48d024e4296b.mp4

#### Release Note
No release note because this does not affect cloud and because it should be cherry-picked to the latest release candidate prior to its release.

```release-note
NONE
```

### CODE REVIEW

**Comment by amyblais** [comment]
@mkraft @pvev Please help prioritize PR reviews on this today if you can.

**Comment by neallred** [comment]
@stevemudie looks like it wasn't picking up the latest server changes for some reason, but I think it has them now. Can you re-verify?

**Review by stevemudie — APPROVED** [review]
Looks good @neallred!

File: components/preparing_workspace/preparing_workspace.tsx [hunk 1]
```diff
@@ -166,6 +167,24 @@ export default function PreparingWorkspace(props: Props) {
...
+                setForm({
+                    ...form,
+                    plugins: {
+                        skipped: false,
+                    },
+                });
+            }
+            if (currentStep === WizardSteps.Plugins) {
+                const mostRecentStepIndex = stepOrder.indexOf(mostRecentStep);
+                setStepHistory([mostRecentStep, stepOrder[mostRecentStepIndex - 1]]);
```
[the remark this answers is not in the document]
Comment (neallred) [reply components/preparing_workspace/preparing_workspace.tsx:183]: Good question! I don't think so, but just to be safe, I should make sure its never less than 0.

**Review by nickmisasi — APPROVED** [review]

**Pull request merged**